### ISSUE TITLE
Add ability to get the server time

### ISSUE THREAD

**Issue by mmacfadden** [convergencelabs/convergence-project#241]
**Versions**
 - Convergence Version: 1.0.0-rc.9

**Description**
Operations and other events are generally timestamped with the server's impression of the time.  It would be usefull to allow API consumers to query the server to understand what its time is (relative to their own) so they can properly understand the times. This is potentially important if the servers time and the remote clients times are not in sync.

It is understood that the network latency between the server and client will cause drift, but the drift will hopefully be roughy the same for the "get time" call as well as the other messages.

The proposed API is as follows:

```ts
class ConvergenceDomain {
  public serverTime(): Promise<Date>;
}
```

Two new protocol messages will be added:

```proto
message GetServerTimeRequestMessage {
  option (scalapb.message).extends = "com.convergencelabs.convergence.proto.RequestMessage";
  option (scalapb.message).extends = "com.convergencelabs.convergence.proto.ClientMessage";
}

message GetServerTimeResponseMessage {
  option (scalapb.message).extends = "com.convergencelabs.convergence.proto.ResponseMessage";
  option (scalapb.message).extends = "com.convergencelabs.convergence.proto.ServerMessage";

  google.protobuf.Timestamp serverTime = 1;
}
```

They will be added to the ConvergenceMessage body one of.

```proto
message ConvergenceMessage {
  ...

  oneof body {
    ...

    core.GetServerTimeRequestMessage serverTimeRequest = 20;
    core.GetServerTimeResponseMessage serverTimeResponse = 21;

    ...
  }
}
```

**Comment by toebes** [comment]
Does the serverTime() method result in a message going back to the server?  Assuming that is the case, what happens if the server has been disconnected?

**Comment by mmacfadden** [comment]
@toebes Good question.  I am not sure how to answer the question of the server time, if we are not connected to the sever. I assume you are asking if we WERE connected and then got disconnected.  The only way to really do that would be to get the server time proactively. Compare that to the local browsers time and create a delta, and then answer that question from there.  I don't really see that as core Convergence functionality, though and would leave that up to the developer to manage.

**Comment by mmacfadden** [comment]
This has been implemented and will be released in 1.0.0-rc.10
